test/rpl-usdc-testing.js

```diff
@@ -116,23 +116,46 @@ describe("RPL-USDC Pool Testing", function () {
   }
 
   it("Test core functions", async function () {
-    const [USDC, RPL, pool, deployer, lp1, lp2, lp3, borrower1, borrower2] = await setup()
+    const [USDC, RPL, pool, deployer, lp1, lp2, lp3, borrower1, newCreator] = await setup()
 
     // check balances pre
     const lp1UsdcPreAddCheck = await USDC.balanceOf(lp1.address)
+    const lp2UsdcPreAddCheck = await USDC.balanceOf(lp2.address)
     const poolUsdcPreCheck = await USDC.balanceOf(pool.address)
 
-    // add liquidity
+    // lpwhitelist checks and add liquidity
     const addAmount = ONE_USDC.mul(100000)
-    const blocknum = await ethers.provider.getBlockNumber();
-    const timestamp = (await ethers.provider.getBlock(blocknum)).timestamp;
-    await pool.connect(lp1).addLiquidity(lp1.address, addAmount, timestamp+60, 0);
-    
+    const blocknum = await ethers.provider.getBlockNumber()
+    const timestamp = (await ethers.provider.getBlock(blocknum)).timestamp
+    // check adding only possible if on lpwhitelist
+    await expect(pool.connect(lp1).addLiquidity(lp1.address, addAmount, timestamp+60, 0)).to.be.revertedWithCustomError(pool, "UnapprovedSender")
+    await expect(pool.connect(lp1).toggleLpWhitelist(lp1.address)).to.be.revertedWithCustomError(pool, "UnapprovedSender")
+    await expect(pool.connect(lp2).toggleLpWhitelist(lp1.address)).to.be.revertedWithCustomError(pool, "UnapprovedSender")
+    await pool.connect(deployer).toggleLpWhitelist(lp1.address)
+    await pool.connect(lp1).addLiquidity(lp1.address, addAmount, timestamp+60, 0)
```

Comment:
should be fine, but maybe add test case after you are whitelisted that you still can't add on someone else's behalf
`await expect(pool.connect(lp1).addLiquidity(lp2.address, addAmount, timestamp+60, 0)).to.be.revertedWithCustomError(pool, "UnapprovedSender")`

Reply:
good catch, adding now